rules/testutils/problem_stub.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutils
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/jhump/protoreflect/desc"
+)
+
+// ProblemStub describes a subset of a Problem, and is used to verify
+// that problems look correct in tests.
+type ProblemStub struct {
+	Descriptor desc.Descriptor
+	Message    string
+}
+
+// VerifyDescriptor verifies that the stub's descriptor and
+// problem's descriptor match.
+func (ps *ProblemStub) VerifyDescriptor(p lint.Problem, t *testing.T) {
+	if !reflect.DeepEqual(p.Descriptor, ps.Descriptor) {
```

Comment:
Remind me again why cmp can't be used?

Reply:
Descriptor has unexpected properties. I think I can get it to work, though, let me look into it. (I could not elsewhere because of a circular import but that is not an issue here.)